### PR TITLE
Fixes e-bolas being unblockable and not ensnaring flying mobs.

### LINKED ISSUE #56442: energy bolas try to ensare you before calling ..()
```/obj/item/restraints/legcuffs/bola/energy/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
    if(iscarbon(hit_atom))
        var/obj/item/restraints/legcuffs/beartrap/B = new /obj/item/restraints/legcuffs/beartrap/energy/cyborg(get_turf(hit_atom))
        B.Crossed(hit_atom)
        qdel(src)
    ..()
```
This could cause them to ignore block chances or something (do normal bolas do that?), idk.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Title. Plus reduced indentation.

## Why It's Good For The Game
This will fix #56442.

## Changelog
:cl:
fix: Fixed e-bolas being unblockable, and not being able to ensnare flying/floating mobs (though kinda pointless because they ignore the slowdown).
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
I remember a 2nd PR on this--I deem not being able to catch the e-bola as a feature, if your PR alters that behavior, can you change that?

**Comment by ATH1909** [comment]
> I remember a 2nd PR on this--I deem not being able to catch the e-bola as a feature, if your PR alters that behavior, can you change that?

https://github.com/tgstation/tgstation/pull/58460

**Comment by Ghommie** [comment]
> 
> I deem not being able to catch the e-bola as a feature

Give me one good reason why a parent call being wrongfully


> 
> 
> I remember a 2nd PR on this--I deem not being able to catch the e-bola as a feature, if your PR alters that behavior, can you change that?

Change what exactly? It currently calls ensnare() after being queued for deletion on throw_impact(), and ensnare() won't do jackshit because the mob is already ensnared by a previously spawned bear trap. It's shitcode imho. But even if it weren't, I'd still see no reason why they should be uncatchable and unblockable.

**Comment by Mothblocks** [comment]
Just making sure since a previous PR made it so you can catch them, which I didn't like.

**Comment by LemonInTheDark** [comment]
> Fixes e-bolas ignoring block chances and **throw catches**

What does this mean?

**Comment by Ghommie** [comment]
> 
> 
> > Fixes e-bolas ignoring block chances and **throw catches**
> 
> What does this mean?

Rewrote the changelong entry to be more understandable.

**Comment by LemonInTheDark** [comment]
@Mothblocks I think he's making it catchable

**Comment by Mothblocks** [comment]
bro

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Being uncatchable is not a bug. Keep it.

**Comment by moocowswag** [comment]
What about making it so only things that can block energy projectiles can block it? Like the reflective trenchcoat?

**Comment by oklews** [comment]
It's still a physical object, and you shouldn't be able to ignore the most commonly used bola by wearing armor. Block chance is the way to go IMO.

File: code/game/objects/items/handcuffs.dm [hunk 1]
```diff
@@ -477,6 +477,10 @@
 	breakouttime = 6 SECONDS
 	custom_price = PAYCHECK_HARD * 0.35
 
+/obj/item/restraints/legcuffs/bola/energy/Initialize()
+	. = ..()
+	ADD_TRAIT(src, TRAIT_UNCATCHABLE, TRAIT_GENERIC) /// People said energy bolas being uncatchable is a feature.
```
Comment (Mothblocks) [inline code/game/objects/items/handcuffs.dm:482]:
```suggestion
	ADD_TRAIT(src, TRAIT_UNCATCHABLE, TRAIT_GENERIC) // People said energy bolas being uncatchable is a feature.
```

`///` == doc comment

**Pull request merged**